app/controllers/concerns/verify_sp_attributes_concern.rb

```diff
@@ -28,6 +28,7 @@ def update_verified_attributes
       verified_attributes: sp_session[:requested_attributes],
       last_consented_at: Time.zone.now,
       clear_deleted_at: true,
+      email_address_id: user_session[:selected_email_id],
```

Comment:
On the topic of the previous comment:

- What's the difference between this code and the code in the SAML/OIDC controllers? Do we need both?
- Should this be checking the feature flag? Alternatively, should we feature-flag access to the controller which assigns the session value?
- Should it be be falling back to `last_sign_in_email_address` ? (I'm not sure about this one, maybe it _should_ be `nil` if the session value isn't assigned?)

Reply:
It appears that we don't need that in the verify_sp_attributes_concern. I took it out.

I'll look for where the feature flag needs to be set, next thing.

I agree, it should stay nil unless the user is consenting to share. I have that change checked in now.

Reply:
With taking email_address_id out of the verify_sp_attributes_concern feature flag seems to be covering any related changes.